pkg/containertools/untarer_test.go

```diff
@@ -0,0 +1,116 @@
+package containertools
+
+import (
+	"archive/tar"
+	"context"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"reflect"
+	"sync"
+	"testing"
+
+	"github.com/sirupsen/logrus"
+)
+
+var files = map[string]string{
+	"index.html": `<body>Hello!</body>`,
+	"lang.json":  `[{"code":"eng","name":"English"}]`,
+	"songs.txt":  `Claire de la lune, The Valkyrie, Swan Lake`,
+}
```

Comment:
(nothing blocking obviously) this kind of structure feels like it could be baked into a table driven test instead.